### ISSUE TITLE
New Component: Step indicator

### ISSUE THREAD

**Issue by suzubara** [trussworks/react-uswds#487]
# [Step indicator](https://designsystem.digital.gov/components/step-indicator/)

A step indicator updates users on their progress through a multi-step process.

![image](https://user-images.githubusercontent.com/2723066/94746951-7e349700-0343-11eb-91ef-aca9580542a2.png)

![image](https://user-images.githubusercontent.com/2723066/94746976-8c82b300-0343-11eb-98f7-c3d42a6e70d1.png)

## Description

A list of steps or screens that indicate a user's progress through a form or process.

> Use the step indicator when the user is working through a form or process that will span several different pages that can be organized into three or more high-level steps or chapters. This helps users understand the progression of long forms and reduces uncertainty as they complete each screen.

> The step indicator is designed to complement standard back/next navigation in a linear sequence, not to be navigation of its own.

## Requirements (proposed)

**Prop Interface**
``` 
StepIndicatorStepProps = {
  status: 'complete' | 'current' | 'incomplete' // default to incomplete
  label: React.ReactNode // required
}

StepIndicatorProps = {
  displayLabels: boolean // default to true
  centered: boolean // default to false
  counterStyle: true | false | 'small' // default to false
}
```

## Example Implementation
```
<StepIndicator centered="true" counterStyle="small" displayLabels="true">
  <StepIndicatorStep status="complete">Personal information</StepIndicatorStep>
  <StepIndicatorStep status="complete">Household status</StepIndicatorStep>
  <StepIndicatorStep status="current">Supporting documents</StepIndicatorStep>
  <StepIndicatorStep>Signature</StepIndicatorStep>
  <StepIndicatorStep>Review and submit</StepIndicatorStep>
</StepIndicator>
```

**More Details**
  -   `<div>`root element with `aria-label="progress"`
  - navigation / links are not part of this component
  - steps are an ordered list `<ol>` inside the root element

## Other Considerations
- custom header render prop?
- language props for "Step x of x", "completed", "not completed"?

**Comment by hinsxd** [comment]
I am interested in making this. Will return to this issue within a day :)

**Comment by MissyM2** [comment]
Hi, there, I would like to work on this, in addition to the alignment of the GovBanner that I noted before.  Let me know if I can do this.

**Comment by brandonlenz** [comment]
Merged to our `uswds-2.9.0` branch
